packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -121,15 +120,7 @@ export function PrivateBlockToolbar( {
 
 	const isLargeViewport = ! useViewportMatch( 'medium', '<' );
 
-	const isToolbarEnabled =
-		blockType &&
-		hasBlockSupport( blockType, '__experimentalToolbar', true );
-	const hasAnyBlockControls = useHasAnyBlockControls();
-
-	if (
-		! isToolbarEnabled ||
-		( ! isDefaultEditingMode && ! hasAnyBlockControls )
-	) {
+	if ( ! useHasBlockToolbar() ) {
```

Comment:
Maybe
```suggestion
	const hasBlockToolbar = useHasBlockToolbar();
	if ( hasBlockToolbar ) {
```

It's the coding pattern used throughout the codebase.